src/librustc_typeck/check/method/suggest.rs

```diff
@@ -173,7 +173,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                      if mode == Mode::MethodCall {
                                          "method"
                                      } else {
-                                         "associated item"
+                                         match item_name.as_str().chars().next() {
+                                             Some(name) => {
+                                                 if name.is_lowercase() {
+                                                     "function or associated item"
```

Comment:
Isn't the rust nomenclature for associated function "method"? So "method or associated type" would be more precise, since it can't possibly be an associated const.